internal/db/db.go

```diff
@@ -132,8 +137,42 @@ func NewDB(dir, userID string) (*DB, bool, error) {
 	return &DB{db: client}, !exists, nil
 }
 
+// DeleteDB will rename all the database files for the given user to a directory within the same folder to avoid
+// issues with ent not being able to close the database on demand. The database will be cleaned up on the next
+// run on the Gluon server.
 func DeleteDB(dir, userID string) error {
-	return os.Remove(getDatabasePath(dir, userID))
+	// Rather than deleting the files immediately move them to a directory to be cleaned up later.
+	deferredDeletePath := getDeferredDeleteDBPath(dir)
+
+	if err := os.MkdirAll(deferredDeletePath, 0o700); err != nil {
+		return fmt.Errorf("failed to create deferred delete dir: %w", err)
+	}
+
+	matchingFiles, err := filepath.Glob(fmt.Sprintf("%v/%v*", dir, userID))
```

Comment:
would this be better?

```suggestion
	matchingFiles, err := filepath.Glob(filepath.Join(dir, userID))
```

Reply:
Good catch. Fixed.